From 6c7b083bb791bd3e7f8b10c24d7f38aa6b25f6b8 Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Mon, 26 Nov 2012 18:02:46 +0900
Subject: [PATCH 26/39] base_filter: set end of frame flag for buffers
 containing SPS/PPS, in addition to the config codec
 flag

The REL OMXIL specification states that the ENDOFFRAME flag should be set
for buffers containing the SPS/PPS only.
---
 omx/gstomx_base_filter.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/omx/gstomx_base_filter.c b/omx/gstomx_base_filter.c
index 065985f..c6457df 100644
--- a/omx/gstomx_base_filter.c
+++ b/omx/gstomx_base_filter.c
@@ -692,7 +692,8 @@ pad_chain (GstPad * pad, GstBuffer * buf)
         buffer_offset += omx_buffer->nFilledLen;
 
         if (GST_BUFFER_FLAG_IS_SET (buf, GST_BUFFER_FLAG_PREROLL))
-          omx_buffer->nFlags = OMX_BUFFERFLAG_CODECCONFIG;
+          omx_buffer->nFlags =
+              OMX_BUFFERFLAG_CODECCONFIG | OMX_BUFFERFLAG_ENDOFFRAME;
         else
           omx_buffer->nFlags = OMX_BUFFERFLAG_ENDOFFRAME;
 
-- 
1.7.9.5

